File: app/assets/javascripts/events/events.controller.js
angular
    .module('odrEvents')
    .controller("EventController", ['eventSchema', 'odrEventsApi', '$http', '$scope', function(eventSchema, odrEventsApi, $http, $scope){
        $scope.eventSchema = eventSchema;
        $scope.events = odrEventsApi.events;
        $scope.newEvent = {};

        $scope.getAllEvents = function() {
            return odrEventsApi.getAll();
        }
        
        $scope.addEvent = function() {
            odrEventsApi.create({
                host_id: 0, 
                course_id: 0, 
                title: $scope.newEvent.title,
                distance: $scope.newEvent.distance, 
                eventType: $scope.newEvent.type, 
                city: $scope.newEvent.city, 
                state: $scope.newEvent.state, 
                country: $scope.newEvent.country, 
                entryFee: $scope.newEvent.entryFee, 
                description: $scope.newEvent.description
            });
        }
}]);

File: app/controllers/events_controller.rb
class EventsController < ApplicationController

    def as_json(options = {})
        super(options.merge(include: :host))
    end

    def event_params
        params.permit(:host_id, :course_id, :distance, :eventType, :city, :state, :country, :entry_fee, :description, :title);
    end
    
    def index
        respond_with Event.all
    end

    def show
        @event = Event.find(params[:id])
        @course = Course.find(@event.course_id)
        @segments = CourseSegment.find_by({course_id: @course.id})
        @response = {:event => @event, :course => @course, :segments => @segments}
        respond_with @response
    end

    def new
    end

    def edit
    end

    def create
        respond_with Event.create(event_params)
    end

    def update
        respond_with Event.update(event_params)
    end

    def destroy
        respond_with Event.destroy(params[:id])
    end
end
